ingestion_server/ingestion_server/indexer.py

```diff
@@ -533,4 +548,6 @@ def create_and_populate_filtered_index(
 
         self.refresh(index_name=destination_index, change_settings=True)
 
+        if self.progress is not None:
+            self.progress.value = 100  # mark job as completed
```

Comment:
This was missing from the original implementation and is used rather than the `ping_callback` approach by the catalog to check if a task has completed without error.